src/boundaries/stagger_and_step.jl

```diff
@@ -0,0 +1,255 @@
+this next point, this is the _step_ part of the method. If 
+not, the method search randomly the next point in a 
+neighborhood following a given probability distribution, this 
+is the _stagger_ part. This part sometimes fails to find a new 
+candidate and a new starting point of the trajectory is chosen 
+within the defined region. See the keyword argument 
+`stagger_mode` for the different available methods.   
+
+The method produces a pseudo-trajectory of `N` points δ-close
+to the stable manifold of the chaotic saddle. 
+
+"""
+function stagger_and_step(ds::DynamicalSystem, x0, N::Int, isinside::Function; δ = 1e-10, Tm = 30, 
+    γ = 1.1, max_steps = Int64(1e5), max_escape_time = 10000, stagger_mode = :exp, δ₀ = 1.0, 
+    show_progress = true,  rng::AbstractRNG = Xoshiro())
+
+    progress = ProgressMeter.Progress(
+        N; desc = "Saddle estimation: ", dt = 1.0
+    )
+    xi = stagger_trajectory(ds, x0, Tm, isinside; δ₀, stagger_mode = :unif, 
+                            max_steps, γ, max_escape_time, rng) 
+    if isnothing(xi)
+        error("Cannot find a stagger trajectory. Choose a different starting point or 
+                search radius δ₀.")
+    end
+
+    v = Vector{typeof(current_state(ds))}(undef, N)
+    v[1] = xi
+    for n in 1:N
+        show_progress && ProgressMeter.update!(progress, n)
+        if escape_time!(ds, xi, isinside; max_escape_time) > Tm
+            reinit!(ds, xi)
+        else
+            xp, Tp = stagger!(ds, xi, δ, Tm, isinside; stagger_mode, max_steps, γ, max_escape_time, rng)
+            # The stagger step may fail. We reinitiate the algorithm from a new initial condition.
+            if Tp < 0
+                xp = stagger_trajectory(ds, x0, Tm, isinside; δ₀, stagger_mode = :exp,max_steps, γ, max_escape_time, rng) 
+                if isnothing(xp)
+                    error("Cannot find a stagger trajectory. Choose a different starting 
+                          point or search radius δ₀.")
+                end
+                δ = 0.1
+            end
+            reinit!(ds, xp)
+        end 
+        step!(ds)
+        xi = copy(current_state(ds))
+        v[n] = xi
+    end
+    return StateSpaceSet(v)
+end
+
+
+"""
+    stagger_trajectory(ds, x0, Tm, isinside; kwargs...) -> xi
+
+On success, the function returns a point `xi` with the property `T(xi) > 
+Tm` with a random walk search around the initial coordinates 
+`x0`. `T(xi)` is the escape time of the initial condition `xi` from 
+the bounding box defined by the function `isinside`.  
+In the case where the algorithm cannot find a suitable point, the algorithm 
+returns nothing. 
+
+This is an auxiliary function for [`stagger_and_step`](@ref). 
+Keyword arguments and definitions are identical for both functions. 
+"""
+function stagger_trajectory(ds, x0, Tm, isinside; δ₀ = 1., stagger_mode = :exp,
+        max_steps = Int(1e5), γ = 1.1, max_escape_time = 10000, rng::AbstractRNG = Xoshiro())
```

Comment:
Defaults should only be given at one place. For easier maintainance, you shouldn't specify `max_escape_time = 10000` in three functions. You should only specify it at the top-level function. The rest of the functions should either have `max_escape_time` as a positiononal arguemnt, or in this case, as a mandatory keyword argument. This goes for all other keywords.